decomposer/neuro_san_solver.py

```diff
@@ -0,0 +1,383 @@
+# Copyright © 2025 Cognizant Technology Solutions Corp, www.cognizant.com.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# END COPYRIGHT
+
+import logging
+import os
+import re
+import sys
+import threading
+
+from neuro_san.client.agent_session_factory import AgentSession
+from neuro_san.client.streaming_input_processor import StreamingInputProcessor
+
+from decomposer.session_manager import SessionManager
+from decomposer.solver import Solver
+
+LOG_LEVEL = os.getenv("LOG_LEVEL", "INFO").upper()
+logging.basicConfig(level=LOG_LEVEL, format="[%(levelname)s] %(message)s", stream=sys.stderr)
+
+_DECOMP_FIELD_RE = re.compile(r"(P1|P2|C)\s*=\s*\[(.*?)]", re.DOTALL)
+
+os.environ["AGENT_MANIFEST_FILE"] = "./registries/manifest.hocon"
+os.environ["AGENT_TOOL_PATH"] = "coded_tools"
+
+# Tuning knobs with environment variable overrides
+WINNING_VOTE_COUNT = int(os.getenv("WINNING_VOTE_COUNT", "2"))
+CANDIDATE_COUNT = (2 * WINNING_VOTE_COUNT) - 1
+NUMBER_OF_VOTES = (2 * WINNING_VOTE_COUNT) - 1
+SOLUTION_CANDIDATE_COUNT = (2 * WINNING_VOTE_COUNT) - 1
+
+
+class NeuroSanSolver(Solver):
+    """
+    Generic solver implementation that uses Neuro SAN.
+    """
```

Comment:
This is where all the stuff deleted from mutliagent_reasoner.py comes
The methods are now all part of a class.

This is nice because this is likely where I start making a neuro-san CodedTool for reuse.

Future refactors might make start to make things member variables.
Not for this PR.